zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/MigrateProcessInstanceTest.java

```diff
@@ -1091,6 +1094,164 @@ public void shouldUnsubscribeUnmappedMessageBoundaryEvent() {
+    // so the jobType didn't change
+    ENGINE.job().ofInstance(processInstanceKey).withType("jobTypeA").complete();
+
+    assertThat(
+            RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
+                .withProcessInstanceKey(processInstanceKey)
+                .withElementType(BpmnElementType.END_EVENT)
+                .withElementId("target_process_end")
+                .findAny())
+        .describedAs("Expect that the process instance is continued in the target process")
+        .isPresent();
+  }
+
+  @Test
+  public void shouldWriteMigratedEventWhenActiveElementHasAProcessIncident() {
+    // given
+    final String sourceProcessId = "process";
+    final String targetProcessId = "process2";
+    final var deployment =
+        ENGINE
+            .deployment()
+            .withXmlResource(
+                Bpmn.createExecutableProcess(sourceProcessId)
+                    .startEvent()
+                    .serviceTask(
+                        "A",
+                        b ->
+                            b.zeebeJobType("jobTypeA")
+                                .zeebeInputExpression("assert(x, x != null)", "y"))
+                    .endEvent()
+                    .done())
+            .withXmlResource(
+                Bpmn.createExecutableProcess(targetProcessId)
+                    .startEvent()
+                    .serviceTask("B", t -> t.zeebeJobType("jobTypeB"))
+                    .endEvent("target_process_end")
+                    .done())
+            .deploy();
+    final long processInstanceKey =
+        ENGINE.processInstance().ofBpmnProcessId(sourceProcessId).create();
+
+    final Record<IncidentRecordValue> incident =
+        RecordingExporter.incidentRecords(IncidentIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .getFirst();
+
+    final long targetProcessDefinitionKey =
+        extractProcessDefinitionKeyByProcessId(deployment, targetProcessId);
+
+    // when
+    ENGINE
+        .processInstance()
+        .withInstanceKey(processInstanceKey)
+        .migration()
+        .withTargetProcessDefinitionKey(targetProcessDefinitionKey)
+        .addMappingInstruction("A", "B")
+        .migrate();
+
+    // then
+    assertThat(
+            RecordingExporter.incidentRecords(IncidentIntent.MIGRATED)
+                .withProcessInstanceKey(processInstanceKey)
+                .getFirst()
+                .getValue())
+        .describedAs("Expect that process definition is updated")
+        .hasProcessDefinitionKey(targetProcessDefinitionKey)
+        .hasBpmnProcessId(targetProcessId)
+        .describedAs("Expect that element id changed due to mapping")
+        .hasElementId("B");
+
+    // after resolving the incident, the process should continue as expected
+    ENGINE.variables().ofScope(processInstanceKey).withDocument(Map.of("x", 1)).update();
```

Comment:
Yeah, that is actually correct. I missed that. But interestingly, the error message that is propagated to the newly created instance will be wrong because the incident will not be valid anymore. Maybe we can retry incidents after migrating them? I do not think it is needed for this iteration but something to consider for better UX. @korthout

Reply:
🤔 Interesting point. We can offer this through the API at some point, but by we'll preserve the instance as much as possible, so we shouldn't try resolving the incident automatically by default. We could consider updating the incident error message text if we believe that makes sense. But let's keep it out of scope for now

Reply:
Yeah, I agree. I think, the later option can be tricky since it will probably require to run a fake incident resolve command to extract the updated error message. But definitely out of scope for now. 👍